pkg/rbac/userData.go

```diff
@@ -37,20 +35,37 @@ type userData struct {
 	authzClient v1.AuthorizationV1Interface
 }
 
-func (cache *Cache) GetUserData(ctx context.Context, clientToken string,
-	authzClient v1.AuthorizationV1Interface) (*userData, error) {
-	var user *userData
-	uid := cache.tokenReviews[clientToken].tokenReview.Status.User.UID //get uid from tokenreview
+// Stuct to keep a copy of users access
+type UserDataCache struct {
+	CsResources     []Resource            // Cluster-scoped resources on hub the user has list access.
+	NsResources     map[string][]Resource // Namespaced resources on hub the user has list access.
+	ManagedClusters []string              // Managed clusters where the user has view access.
+}
```

Comment:
Naming is backwards.  I expect this struct to be `UserData` and the other one `UserDataCache` because it has the added fields t manage the caching.